get_file_copy.go

```diff
@@ -27,3 +28,32 @@ func Copy(ctx context.Context, dst io.Writer, src io.Reader) (int64, error) {
 		}
 	}))
 }
+
+// copyReader copies from an io.Reader into a file, using umask to create the dst file
+func copyReader(dst string, src io.Reader, mode os.FileMode) error {
+	dstF, err := os.OpenFile(dst, os.O_RDWR|os.O_CREATE|os.O_TRUNC, mode)
+	if err != nil {
+		return err
+	}
+	defer dstF.Close()
+
+	_, err = io.Copy(dstF, src)
+	return err
+}
+
+// copyFile copies a file in chunks from src path to dst path, using umask to create the dst file
+func copyFile(ctx context.Context, dst string, src string, mode os.FileMode) (int64, error) {
```

Comment:
Let's unit test these 2 funcs.

Reply:
I don't think a direct unit test of these adds much value, they're consumed by all the other unit tested components. Since they're private, I thought separate tests might end up over-specifying them.